### PR TITLE
FIX: Cordon off .dtseries.json contents

### PR DESCRIPTION
**Pull request by effigies**
First step of https://github.com/nipreps/fmriprep/issues/2294.

### CODE REVIEW

**Comment by effigies** [comment]
@mgxd @oesteban Can I trouble one of you for a review? This is passing and can be seen in action in https://github.com/nipreps/fmriprep/pull/2320.

I'm inclined to make a niworkflows bug-fix release with this in.

File: niworkflows/interfaces/bids.py [hunk 1]
```diff
@@ -653,9 +653,24 @@ def _run_interface(self, runtime):
                 }
             )
             if self._metadata:
-                sidecar = Path(self._results["out_file"][0]).parent / (
-                    "%s.json" % _splitext(self._results["out_file"][0])[0]
-                )
+                out_file = self._results["out_file"][0]
```
Comment (mgxd) [inline niworkflows/interfaces/bids.py:656]:
```suggestion
                out_file = Path(self._results["out_file"][0])
```

File: niworkflows/interfaces/bids.py [hunk 2]
```diff
@@ -653,9 +653,24 @@ def _run_interface(self, runtime):
...
+                # only the existing keys should keep going into them.
+                if out_file.endswith(".dtseries.nii"):
+                    legacy_metadata = {}
+                    for key in ("grayordinates", "space", "surface", "surface_density", "volume"):
+                        if key in self._metadata:
+                            legacy_metadata[key] = self._metadata.pop(key)
+                    if legacy_metadata:
+                        sidecar = Path(self._results["out_file"][0]).parent / (
+                            "%s.json" % _splitext(self._results["out_file"][0])[0]
+                        )
```
Comment (mgxd) [inline niworkflows/interfaces/bids.py:669]:
```suggestion
                        sidecar = out_file.parent / f"{_splitext(str(out_file))[0]}.json"
```

File: niworkflows/interfaces/bids.py [hunk 3]
```diff
@@ -653,9 +653,24 @@ def _run_interface(self, runtime):
...
+                        sidecar.write_text(dumps(legacy_metadata, sort_keys=True, indent=2))
+                # The future: the extension is the first . and everything after
+                out_path = Path(out_file)
```
Comment (mgxd) [inline niworkflows/interfaces/bids.py:672]:
```suggestion
```

File: niworkflows/interfaces/bids.py [hunk 4]
```diff
@@ -653,9 +653,24 @@ def _run_interface(self, runtime):
...
+                sidecar = out_path.parent / f"{out_path.name.split('.', 1)[0]}.json"
```
Comment (mgxd) [inline niworkflows/interfaces/bids.py:673]:
```suggestion
                sidecar = out_file.parent / f"{out_file.name.split('.', 1)[0]}.json"
```

**Review by mgxd — APPROVED** [review]
LGTM, minor suggestions to keep the code more consistent but feel free to discard.

**Comment by effigies** [comment]
Sure. I'd kept the old part the same for the simplicity of the edit, but the test makes changing it safer.

File: niworkflows/interfaces/bids.py [hunk 5]
```diff
@@ -653,9 +653,21 @@ def _run_interface(self, runtime):
...
             if self._metadata:
-                sidecar = Path(self._results["out_file"][0]).parent / (
-                    "%s.json" % _splitext(self._results["out_file"][0])[0]
-                )
+                out_file = Path(self._results["out_file"][0])
+                # 1.3.x hack
+                # For dtseries, we have been generating weird non-BIDS JSON files.
+                # We can safely keep producing them to avoid breaking derivatives, but
+                # only the existing keys should keep going into them.
+                if out_file.endswith(".dtseries.nii"):
```
Comment (mgxd) [inline niworkflows/interfaces/bids.py:661]: whoops missed this
```suggestion
                if out_file.name.endswith(".dtseries.nii"):
```

**Comment by effigies** [comment]
Passes locally now. Thanks for the quick fix.

**Pull request merged**